### PR TITLE
Update Compose to dev10.

### PR DESCRIPTION
**Pull request by zach-klippenstein**
# Description

Dev10 finally brings support for coroutines!

This is the first compose update in a while to actually bump the compiler version
as well as the plugin version, which is why it requires a few more changes
to work.
Fixes # (issue)

## Type of change

Please select at least one

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] API change this is not Backward compatible (fix or feature that would cause existing functionality to not work as expected or change the API contract)
- [ ] This change requires a documentation update
- [x] Dependency update

## Testing Checklist
How was this change tested?

- [ ] Unit Tests
- [x] UI Tests
- [ ] Snapshot Tests (iOS only)

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: kotlin/samples/hello-compose-binding/src/androidTest/java/com/squareup/sample/hellocomposebinding/HelloBindingTest.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ import org.junit.runner.RunWith
 class HelloBindingTest {
 
   // Launches the activity.
-  @Rule @JvmField val scenario = ActivityScenarioRule(HelloBindingActivity::class.java)
+  @Rule @JvmField val composeRule = AndroidComposeTestRule<HelloBindingActivity>()
```
Comment (zach-klippenstein) [inline kotlin/samples/hello-compose-binding/src/androidTest/java/com/squareup/sample/hellocomposebinding/HelloBindingTest.kt:31]: Compose top-level UI test functions now require a `ComposeTestRule` to be used, or they will throw.

File: kotlin/workflow-ui/core-compose/src/main/java/com/squareup/workflow/ui/compose/ComposeViewFactory.kt [hunk 2]
```diff
@@ -84,7 +85,7 @@ internal class ComposeViewFactory<RenderingT : Any>(
     )
 
     // Entry point to the composition.
-    composeContainer.setContent {
+    composeContainer.setContent(Recomposer.current()) {
```
Comment (zach-klippenstein) [inline kotlin/workflow-ui/core-compose/src/main/java/com/squareup/workflow/ui/compose/ComposeViewFactory.kt:88]: Old method is deprecated.

File: kotlin/samples/hello-compose-binding/build.gradle.kts [hunk 3]
```diff
@@ -28,6 +30,9 @@ android {
 }
 
 apply(from = rootProject.file(".buildscript/configure-compose.gradle"))
+tasks.withType<KotlinCompile> {
+  kotlinOptions.apiVersion = "1.3"
+}
```
Comment (zach-klippenstein) [inline kotlin/samples/hello-compose-binding/build.gradle.kts:35]: The compiler the plugin is built against is in some weird liminal state between 1.3 and 1.4, so these warnings are safe to ignore, this suppresses them.

Comment (rjrjr) [reply kotlin/samples/hello-compose-binding/build.gradle.kts:35]: Seems like this comment should be in the gradle file.

Comment (zach-klippenstein) [reply kotlin/samples/hello-compose-binding/build.gradle.kts:35]: Yep

Comment (zach-klippenstein) [reply kotlin/samples/hello-compose-binding/build.gradle.kts:35]: Done.

**Comment by zach-klippenstein** [comment]
Looks like this might be an actual test failure, but the Actions UI isn't even showing a log for this task so I github might just be having issues.

**Comment by zach-klippenstein** [comment]
UI test failure is now: `Caused by: java.net.UnknownHostException: dl.google.com`.

**Pull request merged**